test/recipes/70-test_sslvertol.t

```diff
@@ -34,33 +34,69 @@ my $proxy = TLSProxy::Proxy->new(
     (!$ENV{HARNESS_ACTIVE} || $ENV{HARNESS_VERBOSE})
 );
 
+my @available_tls_versions = ();
+foreach (available_protocols("tls")) {
+    unless (disabled($_)) {
+        note("Checking enabled protocol $_");
+        m|^([a-z]+)(\d)(_\d)?|;
+        my $versionname;
+        if (defined $3) {
+            $versionname = 'TLSProxy::Record::VERS_'.uc($1).'_'.$2.$3;
+            note("'$1', '$2', '$3' => $versionname");
+        } else {
+            $versionname = 'TLSProxy::Record::VERS_'.uc($1).'_'.$2.'_0';
+            note("'$1', '$2' => $versionname");
+        }
+        push @available_tls_versions, eval $versionname;
+    }
+}
+note("TLS versions we can expect: ", join(", ", @available_tls_versions));
+
 #This file does tests without the supported_versions extension.
 #See 70-test_sslversions.t for tests with supported versions.
-#Test 1: Asking for TLS1.4 should pass and negotiate TLS1.2
+#Test 1: Asking for TLS1.4 should pass and negotiate the maximum
+#available TLS version according to configuration
 my $client_version = TLSProxy::Record::VERS_TLS_1_4;
-$proxy->clientflags("-no_tls1_3");
 $proxy->start() or plan skip_all => "Unable to start up Proxy for tests";
 plan tests => 3;
 my $record = pop @{$proxy->record_list};
-ok(TLSProxy::Message->success()
-   && $record->version() == TLSProxy::Record::VERS_TLS_1_2,
-   "Version tolerance test, TLS 1.4");
+ok((note("Record version received: ".
+         (defined $record ? $record->version() : "none")),
+    TLSProxy::Message->success())
+   && $record->version() == tls_version_below(TLSProxy::Record::VERS_TLS_1_4),
```

Comment:
Without supported_versions the max negotiated version should be TLSv1.2...so the above needs to say VERS_TLS_1_3